docs/reference/query-dsl/terms-set-query.asciidoc

```diff
@@ -1,121 +1,220 @@
+two of the following terms:
+
+* `blue`
+* `green`
+* `red`
+
+The `minimum_should_match_field` is `required_matches`. This means the minimum
+number of matching terms required is `2`, the value of the `required_matches`
+field.
 
 [source,js]
---------------------------------------------------
+----
 GET /my-index/_search
 {
     "query": {
         "terms_set": {
-            "codes" : {
-                "terms" : ["abc", "def", "ghi"],
-                "minimum_should_match_field": "required_matches"
+            "color": {
+                "terms": ["blue", "green", "red"],
+                "minimum_should_match_field": "required_matches",
+                "boost": 1.0
             }
         }
     }
 }
---------------------------------------------------
+----
 // CONSOLE
 
-Response:
+[[terms-set-top-level-params]]
+==== Top-level parameters for `terms_set`
 
-[source,js]
---------------------------------------------------
-{
-  "took": 13,
-  "timed_out": false,
-  "_shards": {
-    "total": 1,
-    "successful": 1,
-    "skipped" : 0,
-    "failed": 0
-  },
-  "hits": {
-    "total" : {
-        "value": 1,
-        "relation": "eq"
-    },
-    "max_score": 0.87546873,
-    "hits": [
-      {
-        "_index": "my-index",
-        "_type": "_doc",
-        "_id": "2",
-        "_score": 0.87546873,
-        "_source": {
-          "codes": ["def", "ghi"],
-          "required_matches": 2
-        }
-      }
-    ]
-  }
-}
---------------------------------------------------
-// TESTRESPONSE[s/"took": 13,/"took": "$body.took",/]
+`<field>`::
+Field you wish to search.
+
+[[terms-set-field-params]]
+==== Parameters for `<field>`
+
+`terms`::
++
+--
+Array of terms you wish to find in the provided `<field>`. To return a document,
+a minimum number of terms must exactly match the field values, including
+whitespace and capitalization.
+
+The minimum number of matching terms is defined in the `minimum_should_match_field` or `minimum_should_match_script` parameter.
+--
+
+`minimum_should_match_field`::
+<<number, Numeric>> field containing the minimum number of matching terms
+required to return a document.
+
+`minimum_should_match_script`::
++
+--
+Custom script containing the minimum number of terms to match.
 
-Scripts can also be used to control how many terms are required to match
-in a more dynamic way. For example a create date or a popularity field
-can be used as basis for the number of required terms to match.
+For parameters and valid values, see <<modules-scripting, Scripting>>.
 
-Also the `params.num_terms` parameter is available in the script to indicate the
-number of terms that have been specified.
+For an example query using the `minimum_should_match_script` parameter, see
+<<terms-set-query-script, How to use the `minimum_should_match_script`
+parameter>>.
+--
 
-An example that always limits the number of required terms to match to never
-become larger than the number of terms specified:
+`boost`::
```

Comment:
Like in a previous PR, maybe we could mention the "boost" parameter on all queries for completeness but link to a common paragraph for the detailed discussion.